fast64_internal/oot/oot_cutscene.py

```diff
@@ -0,0 +1,604 @@
+		prop_split(box, listProp, 'fxEndFrame', 'End Frame')
+		return
+	elif listProp.listType == "Lighting":
+		attrName = "lighting"
+	elif listProp.listType == "Time":
+		attrName = "time"
+	elif listProp.listType in ["PlayBGM", "StopBGM", "FadeBGM"]:
+		attrName = "bgm"
+	elif listProp.listType == "Misc":
+		attrName = "misc"
+	elif listProp.listType == "0x09":
+		attrName = "nine"
+	elif listProp.listType == "Unk":
+		prop_split(box, listProp, 'unkType', 'Unk List Type')
+		attrName = "unk"
+	else:
+		raise PluginError("Invalid listType")
+		
+	dat = getattr(listProp, attrName)
+	for i, p in enumerate(dat):
+		p.draw(box, listProp, listIndex, i, objName, collectionType)
+	if len(dat) == 0:
+		box.label(text = "No items in " + listProp.listType + " List.")
+	if listProp.listType == "Textbox":
+		row = box.row(align=True)
+		for l in range(3):
+			addOp = row.operator(OOTCSTextboxAdd.bl_idname, text = 'Add ' + ootEnumCSTextboxType[l][1], icon = ootEnumCSTextboxTypeIcons[l])
+			addOp.collectionType = collectionType + '.textbox'
+			addOp.textboxType = ootEnumCSTextboxType[l][0]
+			addOp.listIndex = listIndex
+			addOp.objName = objName
+	else:
+		addOp = box.operator(OOTCollectionAdd.bl_idname, text = 'Add item to ' + listProp.listType + ' List')
+		addOp.option = len(dat)
+		addOp.collectionType = collectionType + '.' + attrName
+		addOp.subIndex = listIndex
+		addOp.objName = objName
+
+
+class OOTCSListAdd(bpy.types.Operator):
+	bl_idname = 'object.oot_cslist_add'
+	bl_label = 'Add CS List'
+	bl_options = {'REGISTER', 'UNDO'} 
+
+	collectionType : bpy.props.StringProperty()
+	listType : bpy.props.EnumProperty(items = ootEnumCSListType)
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		collection = getCollection(self.objName, self.collectionType, None)
+		collection.add()
+		collection[len(collection)-1].listType = self.listType
+		#self.report({'INFO'}, 'Success!')
+		return {'FINISHED'} 
+
+def drawCSAddButtons(layout, objName, collectionType):
+	def addButton(row):
+		nonlocal l
+		op = row.operator(OOTCSListAdd.bl_idname, text = ootEnumCSListType[l][1], icon = ootEnumCSListTypeIcons[l])
+		op.collectionType = collectionType
+		op.listType = ootEnumCSListType[l][0]
+		op.objName = objName
+		l += 1
+	box = layout.column(align=True)
+	l = 0
+	row = box.row(align=True)
+	row.label(text = 'Add:')
+	addButton(row)
+	for _ in range(3):
+		row = box.row(align=True)
+		for _ in range(3):
+			addButton(row)
+	box.label(text = 'Install zcamedit for camera/actor motion.')
+
+
+class OOTCutsceneProperty(bpy.types.PropertyGroup):
+	csEndFrame : bpy.props.IntProperty(name = "End Frame", min = 0, default = 100)
+	csWriteTerminator : bpy.props.BoolProperty(name = "Write Terminator (Code Execution)")
+	csTermIdx : bpy.props.IntProperty(name = "Index", min = 0)
+	csTermStart : bpy.props.IntProperty(name = "Start Frm", min = 0, default = 99)
+	csTermEnd : bpy.props.IntProperty(name = "End Frm", min = 0, default = 100)
```

Comment:
What is "Frm" ?

Reply:
"Frame". This has to fit within the background of one of those value scroll boxes, next to the value itself, so the characters are very limited.

Reply:
Then what's the difference between "end frame" and "end frm" 🤔

Reply:
`csEndFrame` is the end frame of the cutscene. `csTermStart` and `csTermEnd` are the start and end frames of the terminator. This is clear from the GUI (at least I think it is, if you think there should be an extra box or something around them then I can add that).
Also the `csEndFrame` value scroll box extends across the whole panel, whereas `csTermIdx`, `csTermStart`, and `csTermEnd` share a row. That's why there is enough room to fit more text for "End Frame".

Reply:
![Screenshot_20210910_234609](https://user-images.githubusercontent.com/11657682/132939245-bce41f1b-f730-4122-bfcb-bb9d8c3dc580.png)

Reply:
The terminator shouldn't actually need to be more than one frame to begin with

Reply:
@Zelllll For a normal terminator which loads a new scene, that is correct. However:
1. The game checks both start and end frames, so end frame is not unused. fast64 + zcamedit together are supposed to be able to edit any cutscene data besides unused data.
2. Some vanilla terminators contain custom code. Someone could make a custom terminator which, for example, checks for some condition in some actor doing something in the cutscene before triggering the transition. Then, they would want that terminator to run for multiple frames.